topics/icrc_27_get_icrc_1_subaccounts.md

```diff
@@ -0,0 +1,202 @@
+# ICRC-27: Get ICRC-1 Subaccounts
+
+[![Status Badge](https://img.shields.io/badge/STATUS-DRAFT-ffcc00.svg)](https://github.com/orgs/dfinity/projects/31)
+[![Extension Badge](https://img.shields.io/badge/Extends-ICRC--25-ffcc222.svg)](./icrc_25_signer_interaction_standard.md)
+
+<!-- TOC -->
+* [ICRC-27: Get ICRC-1 Subaccounts](#icrc-27-get-icrc-1-subaccounts)
+  * [Summary](#summary)
+  * [Use-Case](#use-case)
+  * [Method](#method)
+  * [Scope (according to the ICRC-25 standard)](#scope-according-to-the-icrc-25-standard)
+    * [Example Permission Request](#example-permission-request)
+  * [`icrc25_supported_standards`](#icrc25_supported_standards)
+  * [Request](#request)
+    * [Example RPC Request](#example-rpc-request)
+  * [Response](#response)
+    * [Example RPC Response](#example-rpc-response)
+  * [Message Processing](#message-processing)
+  * [Errors](#errors)
+  * [Default subaccount](#default-subaccount)
+  * [Other ledger standards](#other-ledger-standards)
+<!-- TOC -->
+
+## Summary
+
+The purpose of the `icrc27_get_subaccounts` message is for the relying party to receive subaccounts for a given identity
+managed by the signer.
+
+There are many types of signers:
+
+- Signers that support many different standards
+- Signers that support multiple networks
+- Signers that keep a list of subaccounts per ledger
+- Signers that keep a list of subaccounts used across all ledgers
+
+The scope of this spec is limited to `ICRC-1` standard ledgers, the subaccounts returned in this spec contain additional
+optional information `name` and `canisterIds` to indicate with which ledgers the subaccounts are meant to be used.
+
+## Use-Case
+
+The relying party can use the subaccounts in the ledger information to manage funds and assets held by these
+subaccounts.
+
+Example usages:
+
+- Relying party needs to know which subaccount to request a transfer of ICP tokens to
+  make a payment.
+- Relying party needs to know all subaccounts to create tax reports.
+- Relying party needs to know the ckBTC subaccounts to make a ckBTC payment.
+- Relying party needs to know all subaccounts for the ICRC-1 standard to show all tokens the user holds and is able to
+  swap on the dex.
+
+## Method
+
+**Name:** `icrc27_get_subaccounts`
+
+**Prerequisite:** Active session with granted permission scope `icrc27_get_subaccounts` and `icrc31_get_principals`, or `*`.
```

Comment:
My gut feeling tells me, that requiring scopes from other methods is not the right way to go (even if they overlap in functionality). We should definitely discuss this point in the next session.

Reply:
I agree this part is where things got a bit complicated since you can't have a subaccount without a principal, I'll be there in the next meeting (Jan 23th).